### PR TITLE
Bookmark

### PR DESCRIPTION
**Pull request by junaidshaikh-js**
This PR inclues

- user can now bookmark any post
- bookmarked posts can be visible on bookmark page 
- typo and bug fixes

### CODE REVIEW

File: src/features/posts/Post.jsx [hunk 1]
```diff
@@ -54,6 +64,22 @@ export const Post = ({ post }) => {
...
+        await removePostFromBookmark(post.id, token);
+        dispatch(removeFromBookmark(post.id));
+        toast.success("Post removed from bookmark");
+      } else {
+        await bookmarkPost(post.id, token);
+        dispatch(addToBookmark(post.id));
+        toast.success("Post bookmarked");
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (prachi-sahani) [inline src/features/posts/Post.jsx:79]: you can show an error toast instead

**Review by prachi-sahani — COMMENTED** [review]
Overall the changes look fine
Good to merge

File: src/features/posts/Post.jsx (hunk 1 above)
Comment (junaidshaikh-js) [reply src/features/posts/Post.jsx:79]: all the error handling will be done later

**Review by Developers-codz — APPROVED** [review]
Code looks clean UI also look elegant good work on both
Found nothing to suggest, you can merge

**Pull request merged**